Marlin/src/gcode/bedlevel/abl/G29.cpp

```diff
@@ -729,7 +729,6 @@ G29_TYPE GcodeSuite::G29() {
 
           #endif
 
-          abl.reenable = false;
```

Comment:
The reason this line exists is to prevent re-enabling leveling when you have a broken (incomplete) mesh.

Reply:
Yes, but you don't have a broken mesh at this point because the changes are held in local variable `abl`.

Reply:
That's new behaviour in #23868. I should have included this change then.

Reply:
> Yes, but you don't have a broken mesh at this point because the changes are held in local variable abl.

No, there is no temporary mesh being stored in the `abl` variable. The bed mesh is altered directly as it is probed.

Reply:
> No, there is no temporary mesh being stored in the abl variable. The bed mesh is altered directly as it is probed.

Not any more since #23868. Line 733:
```
            abl.z_values[abl.meshCount.x][abl.meshCount.y] = z;
```
and then later on line 808
```
        COPY(bedlevel.z_values, abl.z_values);
```

This was so that `G29 D` would not alter the mesh, as documented.

Reply:
I see, then technically it's alright in its current form.

I didn't realize the new code was adding so much extra SRAM cost to `PROBE_MANUALLY`, which is meant to be the cheaper option. It would be helpful to provide a switch to use the old in-place modification – for the benefit of tiny boards that really need to squeeze out extra SRAM space.

In general, whenever making any change that will impact SRAM usage more than a little, it's good practice to make the extra cost optional, rather than impose it. That way we can still move forward and add new intelligence using large buffers on more powerful boards, but still keep old configurations' build sizes from continually increasing in size.

Reply:
That makes sense to me, especially as I started this 3D printing lark with a 128k Sanguino and even now am still using an ATmega2560 based board (the AVR architecture is quite dear to my heart).

How about making the dry run functionality optional? I can't think of a way of offering dry run that actually works but does not end up temporarily holding two copies of the mesh.

Reply:
Whilst we're at it, I don't think we actually need to store abl.eqnAMatrix. It just stores (x, y, 1) for each probe point and later uses the x and y to build a vector (x, y, 0). So the 1 is redundant. And the x and y can be calculated when abl.eqnAMatrix is used to print the topography map. That would save 3 * GRID_MAX_POINTS_X * GRID_MAX_POINTS_Y * sizeof(float) bytes if PROBE_MANUALLY is used with AUTO_BED_LEVELING_LINEAR).

Reply:
@thinkyhead I'll attack these if you would like me to.